lib/publiccloud/ec2.pm

```diff
@@ -17,13 +17,60 @@ use testapi;
 
 has ssh_key      => undef;
 has ssh_key_file => undef;
+has credentials  => undef;
+
+sub create_credentials {
+    my ($self) = @_;
+
+    if (!defined($self->credentials)) {
+        my $credentials = $self->do_rest_api('/ec2/users', method => 'post');
+        die("Missing key") if (!defined($credentials->{keys}));
+        for my $key (@{$credentials->{keys}}) {
+            next if (!defined($key->{secret}));
+            next if ($key->{status} ne 'active');
+            $self->key_id($key->{key_id});
+            $self->key_secret($key->{secret});
+        }
+        $self->credentials($credentials);
+        die('Failed to create credentials') unless (defined($self->key_id) && defined($self->key_secret));
+    }
+
+    return $self->credentials;
+}
+
+sub delete_credentials {
+    my ($self) = @_;
+
+    return unless (defined($self->credentials));
+
+    my $username = $self->credentials->{name};
+    $self->do_rest_api('/credentials/users/' . $username, method => 'delete');
```

Comment:
```suggestion
     $self->do_rest_api('/credentials/users/' . $self->credentials->{name}, method => 'delete');
```

Reply:
i would keep it, as it made the length less long.